controllers/useraccount/console_usersettings_test.go

```diff
@@ -0,0 +1,449 @@
+			require.Error(t, err)
+			require.Equal(t, "error in getting Role", err.Error())
+			require.Nil(t, result)
+		})
+	})
+
+	t.Run("Get RoleBinding", func(t *testing.T) {
+		t.Run("RoleBinding found by name returns expected object", func(t *testing.T) {
+			rb := &rbac.RoleBinding{
+				TypeMeta: metav1.TypeMeta{Kind: "RoleBinding"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-rolebinding",
+					Namespace: UserSettingNS,
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(rb).Build()
+			name := "test-rolebinding"
+			get_rb := &rbac.RoleBinding{TypeMeta: metav1.TypeMeta{Kind: "RoleBinding"}}
+
+			result, err := getConsoleUserSettingObjectByName(ctx, cl, name, get_rb)
+			require.NoError(t, err)
+			require.NotEmpty(t, result)
+			require.Equal(t, name, result.GetName())
+		})
+		t.Run("RoleBinding not found by name returns nil", func(t *testing.T) {
+			cl := fake.NewClientBuilder().WithObjects().Build()
+			name := "non-existent-rolebinding"
+			rb := &rbac.RoleBinding{TypeMeta: metav1.TypeMeta{Kind: "RoleBinding"}}
+
+			result, err := getConsoleUserSettingObjectByName(ctx, cl, name, rb)
+			require.NoError(t, err)
+			require.Nil(t, result)
+		})
+		t.Run("error in getting RoleBinding", func(t *testing.T) {
+			cl := test.NewFakeClient(t)
+			cl.MockGet = func(ctx context.Context, key client.ObjectKey, obj client.Object, opts ...client.GetOption) error {
+				return fmt.Errorf("error in getting RoleBinding")
+			}
+			name := "error-rolebinding"
+			rb := &rbac.RoleBinding{TypeMeta: metav1.TypeMeta{Kind: "RoleBinding"}}
+
+			result, err := getConsoleUserSettingObjectByName(ctx, cl, name, rb)
+			require.Error(t, err)
+			require.Equal(t, "error in getting RoleBinding", err.Error())
+			require.Nil(t, result)
+		})
+	})
+
+	t.Run("Getting any other object returns error", func(t *testing.T) {
+		secret := &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{Kind: "Secret"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-secret",
+				Namespace: UserSettingNS,
+			},
+		}
+		cl := fake.NewClientBuilder().WithObjects(secret).Build()
+		name := "test-secret"
+		get_obj := &corev1.Secret{TypeMeta: metav1.TypeMeta{Kind: "Secret"}}
+
+		result, err := getConsoleUserSettingObjectByName(ctx, cl, name, get_obj)
+		require.Error(t, err)
+		require.Nil(t, result)
+		require.Equal(t, "object type Secret is not a console setting supported object", err.Error())
+	})
+}
+
+func TestDeleteConsoleSettingObjects(t *testing.T) {
+	t.Run("Delete ConfigMap", func(t *testing.T) {
+		t.Run("Configmap found by name and deleted", func(t *testing.T) {
+			ctx := context.Background()
+			cm := &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{Kind: "ConfigMap"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "user-settings-johnsmith",
+					Namespace: UserSettingNS,
+				},
+			}
+			cl := fake.NewClientBuilder().WithObjects(cm).Build()
+
+			err := deleteConfigMap(ctx, cl, "johnsmith")
+			require.NoError(t, err)
```

Comment:
I saw that you check the abscence of these objects in the useraccount_controller_test but would it be worth checking that the config map is gone here as well ?

Reply:
done in [d4bc88c](https://github.com/codeready-toolchain/member-operator/pull/568/commits/d4bc88cc583bfad7f84185b5fa341ffc508984e3)